common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -288,10 +305,27 @@ bool base_wlan_hal_dwpal::fsm_setup()
                     LOG(ERROR) << "Can't get event fd since dwpal ctx is NULL!";
                     return (transition.change_destination(dwpal_fsm_state::Detach));
                 }
-
                 // Success
                 LOG(DEBUG)
                     << "Open and attach an event interface to wpa_supplicant/hostapd - SUCCESS!";
+
+                // Get the nl event interface file descriptor
+                if (m_dwpal_nl_ctx != nullptr) {
```

Comment:
no entirely true,
unless we add attach = false in case of nl attach failure we will reach this code.

Also, its a good  protection in case we decide not to support NL in other platforms.

Reply:
> no entirely true,
> unless we add attach = false in case of nl attach failure we will reach this code.

In the current code we go to Detach state on failure, so we can't get into this state.

Reply:
Since there is no point to continue if NL is not supported  i suggest 
1. add attach = false in case of nl attach failure (see my [comment](https://github.com/prplfoundation/prplMesh/pull/619#discussion_r362424962))
2. leave this check as is .

Reply:
@arnout is right, but we don't actually check make sure that this variable is not null.
We only move to Detach if `dwpal_driver_nl_attach` returns an error...
Theoretically, since dwpal is an external library, the call may succeed by the variable will point to NULL.
We should simply move this check to the Init state.